pkg/state/transaction_checker.go

```diff
@@ -333,6 +336,138 @@ func (tc *transactionChecker) checkTransferWithSig(transaction proto.Transaction
 	return smartAssets, nil
 }
 
+func (tc *transactionChecker) checkEthereumTransactionWithProofs(transaction proto.Transaction, info *checkerInfo) ([]crypto.Digest, error) {
+	metamaskActivated, err := tc.stor.features.newestIsActivated(int16(settings.RideV6))
+	if err != nil {
+		return nil, errors.Errorf("failed to check whether feature %d was activated: %v", settings.RideV6, err)
+	}
+	if !metamaskActivated {
+		return nil, errors.Errorf("failed to handle ethereum transaction: feature %d has not been activated yet", settings.RideV6)
+	}
+
+	tx, ok := transaction.(*proto.EthereumTransaction)
+	if !ok {
+		return nil, errors.New("failed to convert interface to TransferWithSig transaction")
```

Comment:
Copy-pasted error message

Reply:
Fixed and removed. You can find this check in 'EthereumTransaction.Validate' method.